src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -50,12 +50,26 @@ public Accessor(SQLitePersistentStorage storage)
             protected abstract void BindFirstParameter(SqlStatement statement, TDatabaseId dataId);
             protected abstract TWriteQueueKey GetWriteQueueKey(TKey key);
 
-            public async Task<Stream> ReadStreamAsync(TKey key, CancellationToken cancellationToken)
+            public async Task<Checksum> ReadChecksumAsync(TKey key, CancellationToken cancellationToken)
             {
-                // Note: we're technically fully synchronous.  However, we're called from several
-                // async methods.  We just return a Task<stream> here so that all our callers don't
-                // need to call Task.FromResult on us.
+                using (var stream = await ReadBlobColumnAsync(key, ChecksumColumnName, checksumOpt: null, cancellationToken).ConfigureAwait(false))
+                using (var reader = ObjectReader.TryGetReader(stream, cancellationToken))
+                {
+                    if (reader != null)
+                    {
+                        return Checksum.ReadFrom(reader);
```

Comment:
it would be nice to avoid this allocation here.  @heejaechang Are you ok with Sha1Hash being exposed as a type other components can use directly?

Reply:
Also, is there a reason for there to be a Checksum class at all?  It just seems to wrap a Sha1Hash (and is sealed), so it's unclear to me what purpose it serves.  Can you clarify?  Thanks!

Reply:
There are a very large number of `Checksum` instances, each of which is 24 bytes. Each reference to a `Checksum` is 4 bytes. It would only take a small number of cases where a checksum instance is referenced multiple times for inlining the type to increase overall memory usage. The wrapper should not be removed until a test case is constructed _proving_ that this situation will not cause a problem at scale.